### DISCUSSION TITLE
请问这样设置墙外走代理，墙内直连对吗？

### DISCUSSION THREAD

**Discussion by guanjian1212** [v2fly/v2ray-core#2429] [Q&A]
安卓手机，设置了VPN模式，规则选择了“绕过局域网和及大陆地址而后代理”，这样是不是打开B站APP就不会走代理，浏览器打开谷歌地址就会走代理？

**Comment by lhbdhr** [comment]
按理说是这样，但是会判断不准确，我打开v2rayn之后，访问京东就会很慢，商品图片加载不出来。

**Reply by mydogshitgold** [reply]
Most Android clients support per-app proxies, which should be more precise than geo assets.

**Reply by lhbdhr** [reply]
Indeed.

**Comment by qianbinbin** [comment]
你已经说了“绕过局域网和及大陆地址而后代理”，俗称白名单模式，这也意味着一些冷门境内网站和没被墙的境外网站也会走代理，适合主要用墙外网站的用户
还有一种俗称黑名单模式，GFW 会走代理，其它直连，这也意味着冷门境外网站也会直连，然后访问失败或很慢，毕竟 GFW 是民间建立的，不可能收录所有被墙网站，所以适合主要用墙内网站的用户

**Reply by guanjian1212** [reply]
讲得很清楚，非常感谢！
